src/scanner-cli.ts

```diff
@@ -108,12 +114,19 @@ export async function runScannerCli(
   binPath: string,
 ) {
   log(LogLevel.INFO, 'Starting analysis');
+  // We filter out env properties that are passed to the scanner
+  // otherwise, they would supersede the properties passed to the scanner through SONARQUBE_SCANNER_PARAMS
+  const filteredEnvKeys = ENV_TO_PROPERTY_NAME.map(env => env[0]);
+  const filteredEnv = Object.entries(process.env)
+    .filter(([key]) => !filteredEnvKeys.includes(key))
+    .filter(([key]) => !key.startsWith(ENV_VAR_PREFIX));
```

Comment:
Actually, I think we also need to filter out `SONAR_SCANNER_xxx` env vars, otherwise any of these would supersede the value in `SONARQUBE_SCANNER_PARAMS`?

Reply:
I'm actually more and more inclined to pass everything via args in the cmd. I'm afraid if we start with all this ENV juggling it may actually be difficult to maintain in the future, depending on new priorities. We know for sure that cli args are first priority, and will be, so I think it's worth discussing if we want to change the approach. WDYT?

Reply:
We can't pass everything through CLI because some vars (SONAR_TOKEN, at least, potentially it's the only one) would appear in the process list (`ps`) exposing the token to others, whereas it's safe with env var. So we'd need to filter this out from CLI args, and still pass sensitive env vars through env,

I checked the CLI arg limit, and it looks like it shouldn't be a problem.

Since we have a safe way to pass properties for now, and it should work, should we accept this solution, cut the branch so we can close the sprint and consider creating a hardening ticket to improve this later?

Reply:
That's a very good point about the token. 

Ok, let's go ahead. I'm just afraid we are passing params to the cli/engine in a way that is not by design the most prioritary, so I'm afraid we may face similar issues in the future with other scenarios we are not considering right now. 

Ideally, we should have a channel node -> engine/cli where no other source can override param values. But yeah, let's leave this aside for the moment.